codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customize/SerdeDecorator.kt

```diff
@@ -0,0 +1,50 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+package software.amazon.smithy.rust.codegen.client.smithy.customize
+
+import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
+import software.amazon.smithy.rust.codegen.core.rustlang.Feature
+import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
+
+/**
+ * Decorator that adds the `serde-serialize` and `serde-deserialize` features.
+ */
+class SerdeDecorator : ClientCodegenDecorator {
+    override val name: String = "SerdeDecorator"
+    override val order: Byte = -1
+
+    override fun extras(codegenContext: ClientCodegenContext, rustCrate: RustCrate) {
+        fun feature(featureName: String): Feature {
+            return Feature(featureName, false, listOf("aws-smithy-types/$featureName"))
+        }
+        rustCrate.mergeFeature(feature("serde-serialize"))
+        rustCrate.mergeFeature(feature("serde-deserialize"))
+    }
+
+    // I initially tried to implement with LibRsCustomization but it didn't work some how.
+    companion object {
```

Comment:
Here is the text that is going to be included on the lib.rs docs. 
I tried to use the libRsCustomization but somehow it didn't work.

Reply:
What did it look like when it was overriding libRsCustomization?

Reply:
I tried to find the one that didn't work, but it's lost somewhere in the commit history.
However, I was able to make it work!

It looks like this.  

![image](https://github.com/smithy-lang/smithy-rs/assets/68596478/add0bcbb-d570-47bb-b65b-d87c490dd0e3)

I tried to put two hash marks, 
e.g.
```
//! # Enabling Unstable Features
//! ## How to enable `Serialize` and `Deserialize`
```
but I couldn't make it work.

Reply:
I tried to find the commit that didn't work, but it's lost somewhere in the commit history.